paddle/gserver/evaluators/ChunkEvaluator.cpp

```diff
@@ -243,6 +242,53 @@ class ChunkEvaluator : public Evaluator {
     if (tag == tagSingle_) return true;
     return false;
   }
+
+public:
+  // three metrics: precision, recall and F1-score
+  void getNames(std::vector<std::string>* names) {
+    this->storeLocalValues();
+    names->reserve(this->values_.size());
+    for (auto it = this->values_.begin(); it != this->values_.end(); ++it) {
+      names->push_back(this->config_.name() + "." + it->first);
+    }
+  }
+
+  // get value by field name
+  real getValue(const std::string& name, Error* err) const {
+    this->storeLocalValues();
+    std::vector<std::string> buffers;
+    paddle::str::split(name, '.', &buffers);
+    auto it = this->values_.find(buffers[buffers.size() - 1]);
+    if (it == this->values_.end()) {  // not found
+      *err = Error("No such key %s", name.c_str());
+      return 0.0f;
+    }
+
+    return it->second;
+  }
+
+  // get type of evaluator
+  std::string getType(const std::string& name, Error* err) const {
+    this->getValue(name, err);
+    if (!err->isOK()) {
+      return std::string();
+    }
+    return "chunk";
+  }
+
+private:
+  void storeLocalValues() const {
+    CHECK_GT(numOutputSegments_, 0);
+    CHECK_GT(numLabelSegments_, 0);
+    double precision = (double)numCorrect_ / numOutputSegments_;
+    double recall = (double)numCorrect_ / numLabelSegments_;
```

Comment:
Change it to `double recall = !numLabelSegments_  ? 0 : (double)numCorrect_ / numLabelSegments_;`